tests/element_test_vectors.rs

```diff
@@ -67,14 +67,17 @@ fn serialize(format: Format, elements: &[Element]) -> IonResult<Vec<u8>> {
     Ok(buffer)
 }
 
+/// Allow for anything that implements a user-like reader
+type DynamicReader<'a> = Box<dyn IonReader<Item = StreamItem, Symbol = Symbol> + 'a>;
```

Comment:
🗺️ This lets us plug in the `ElementStreamReader` if we want to which is probably not a bad idea for the future (more coverage).